### PR TITLE
[Engine] Fixes null exception in InstancingProcessor

### PR DESCRIPTION
**Pull request by tebjan**
Fixes null exception in InstancingProcessor when no ModelComponent is present

# PR Details

Just a null check.

- [ ] Needs a cherry-pick to master

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: sources/engine/Stride.Engine/Engine/Processors/InstancingProcessor.cs [hunk 1]
```diff
@@ -87,18 +87,19 @@ private void UpdateInstancing(InstancingComponent instancingComponent, Instancin
             if (instancingComponent.Enabled && instancingComponent.Type != null)
             {
                 var instancing = instancingComponent.Type;
+                var modelComponent = instancingData.ModelComponent;
 
                 // Calculate inverse world and bounding box
                 instancing.Update();
 
-                if (instancingData.ModelComponent != null && instancing.InstanceCount > 0)
+                if (modelComponent != null && modelComponent.Model != null && instancing.InstanceCount > 0)
```
Comment (xen2) [inline sources/engine/Stride.Engine/Engine/Processors/InstancingProcessor.cs:95]: `modelComponent?.Model != null`?

Comment (tebjan) [reply sources/engine/Stride.Engine/Engine/Processors/InstancingProcessor.cs:95]: yes, that is indeed much shorter in terms of changes. But some lines got a little shorter with the local variable. just a matter of taste. if you prefer the `?` operator I'll change it.

Comment (vdvman1) [reply sources/engine/Stride.Engine/Engine/Processors/InstancingProcessor.cs:95]: There's no local variable being assigned in that comparison
xen2 was saying that `modelComponent != null && modelComponent.Model != null` can be written as `modelComponent?.Model != null`
No behaviour change there, no change to what variables get assigned

Comment (tebjan) [reply sources/engine/Stride.Engine/Engine/Processors/InstancingProcessor.cs:95]: Oh, i see it now... Thanks! :-)

**Pull request merged**